File: solace/cmd/cooldown/Icespike.java
package solace.cmd.cooldown;

import java.util.*;
import solace.game.*;
import solace.net.*;
import java.io.*;
import solace.util.*;

/**
 * Potency 200 attack spell that heals 5-10% of MP (based on skill level).
 * @author Ryan Sandor Richards
 */
public class Icespike extends CooldownCommand {
  public Icespike(Player p) {
    super("icespike", p);
    setCooldownDuration(CooldownCommand.GLOBAL_COOLDOWN);
    setInitiatesCombat(true);
    setBasePotency(200);
    setSavingThrow("prudence");
  }

  private void healMp (int level) {
    double pctHeal = 0.05 + (0.05 * level / 100.0);
    int mpHeal = (int)(player.getMaxMp() * pctHeal);
    player.setMp(Math.min(
      player.getMaxMp(),
      player.getMp() + mpHeal));
  }

  public boolean execute(int level, Player target) {
    boolean isHit = executeAttack(target);
    if (isHit) {
      healMp(level);
    }
    return isHit;
  }
}

File: game/skills/evocation/cooldowns/icespike.js
'use strict';

/**
 * Potency 200 attack spell that heals 5-10% of MP (based on skill level).
 * @author Ryan Sandor Richards
 */
Commands.addCooldown('icespike', {
  cooldownDuration: Commands.GLOBAL_COOLDOWN,
  initiatesCombat: true,
  basePotency: 70,
  savingThrow: 'prudence',
  run: function (player, target, level, cooldown) {
    var isHit = cooldown.executeAttack(player, target);
    if (isHit) {
      var pctHeal = 0.05 + (0.05 * level / 100.0);
      var mpHeal = parseInt(player.getMaxMp() * pctHeal, 10);
      player.setMp(Math.min(player.getMaxMp(), player.getMp() + mpHeal));
    }
    return isHit;
  }
});
